src/pretix/plugins/socialauth/views.py

```diff
@@ -28,3 +47,37 @@ def oauth_return(request):
         )
         logger.error('Error while authorizing: user has no email address.')
         return redirect('control:auth.login')
+
+
+class LoginState(StrEnum):
+    ENABLED = "enabled"
+    DISABLED = "disabled"
+
+
+class SocialLoginView(AdministratorPermissionRequiredMixin, TemplateView):
+    template_name = 'socialauth/social_auth_settings.html'
+    LOGIN_PROVIDERS = {'mediawiki': False, 'github': False, 'google': False}
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.gs = GlobalSettingsObject()
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
+        if self.gs.settings.get('login_providers', as_type=dict) is None:
+            self.gs.settings.set('login_providers', self.LOGIN_PROVIDERS)
+        context['login_providers'] = self.gs.settings.get('login_providers', as_type=dict)
+        return context
+
+    def post(self, request, *args, **kwargs):
+        login_providers = self.gs.settings.get('login_providers', as_type=dict)
+        for provider in self.LOGIN_PROVIDERS.keys():
+            value = request.POST.get(f'{provider}_login', '').lower()
+            if value not in [s.value for s in LoginState]:
```

Comment:
As state in previous comment, if `LoginState` is a subclass  of `StrEnum`, its member can compare directly with `str`.
So this check can be as simple as `if value not in LoginState`.

![image](https://github.com/user-attachments/assets/7c754c4a-6d90-443c-9bc2-8d1fd5a30b01)

Reply:
@hongquan, it looks like this syntax is only working for Python 3.12 or later.
I checked it on my python3.11 and a online compiler, both raise TypeError: unsupported operand type(s) for 'in': 'str' and 'EnumType'.